### ISSUE TITLE
VideoStreamPlayer doesn´t match Movie Maker FPS

### ISSUE THREAD

**Issue by jorgerosa** [godotengine/godot#69965]
### Godot version

v4.0.beta8.official [45cac42c0]

### System information

Windows 10

### Issue description

Using internal GODOT 4 - Beta 8 - "**Movie Maker**" everything records very well _(in my own limited tests)_, except when recording a scene that has a texture to a mesh that is coming from a SubViewPort that has a VideoStreamPlayer inside. In the recorded AVI file, we can notice that the speed is different _(it is too fast, seems to me, that is 2x faster)_

![GODOT_4_Beta_8_MovieMaker_1](https://user-images.githubusercontent.com/5511928/207096271-0984d0d1-dadc-4806-8f9d-b5cd7285cf63.jpg)

### Steps to reproduce

.

### Minimal reproduction project

.

**Comment by Calinou** [comment]
VideoStreamPlayer doesn't take `Engine.time_scale` into account, and this is likely a similar case here (as `--fixed-fps` influences game speed depending on FPS).

**Comment by jorgerosa** [comment]
[CINEMA.zip](https://github.com/godotengine/godot/files/10212496/CINEMA.zip)
Added a minimal project, however Movie Writer seems to be working fine in this one ... I am lost now ...
- Still sometimes in the editor, this project ( **exactly as the above** ) it runs and other times not ... Just closes the window.
- But it throws no errors in the output console.

**Comment by cyberpsyche** [comment]
As VideoStreamPlayer doesn't take Engine.time_scale into account, my question is how to solve this problem? 

Do I have to modify the VideoStreamPlayer to let the video stream's FPS fit with Engine's time_scale ?  Is there any other solutions to this issue ?

**Comment by Calinou** [comment]
> As VideoStreamPlayer doesn't take Engine.time_scale into account, my question is how to solve this problem?
> 
> Do I have to modify the VideoStreamPlayer to let the video stream's FPS fit with Engine's time_scale ? Is there any other solutions to this issue ?

Unfortunately, I don't see a way to work around this issue. Video playback is hard.

**Comment by akx** [comment]
> VideoStreamPlayer has to handle a lot of things differently to other nodes because it needs to keep audio and video in sync (among other things). Video playback is hard.

As a concession/workaround, maybe we could have a mode for VideoStreamPlayer where it can ignore any audio in the source video, just so it's able to play the video frames with exact timing. (I suppose this would/could unlock arbitrary video seeking/retiming too...)

**Comment by olivergs** [comment]
In my specific case that would work perfectly. I use the videoplayer for complex effects on top of a background, so they do not have audio.

**Comment by akx** [comment]
@olivergs Yep, same here. I ended up only looking at the video as a preview in Godot and doing the postprocessing in ffmpeg. Luckily in my case I could just use an add blend from the Godot output with my background:

```
ffmpeg -i d24amv_v1_bg.mov -i godot-generated.avi -filter_complex "[0]format=gbrp[bg];[1]format=gbrp[fg];[fg][bg]blend=all_mode='addition':all_opacity=1[final];[final]format=yuv420p" output.mp4
```

**Comment by ReidX7** [comment]
Yes. my workaround is use OBS to record video, not movie mode, 
but the new problem is there is audio quality loss when record by OBS, though it's easier to solve.
